compiler/rustc_error_codes/src/error_codes/E0716.md

```diff
@@ -30,8 +30,9 @@ let q = p;
 
 Whenever a temporary is created, it is automatically dropped (freed) according
 to fixed rules. Ordinarily, the temporary is dropped at the end of the enclosing
-statement -- in this case, after the `let`. This is illustrated in the example
-above by showing that `tmp` would be freed as we exit the block.
+statement -- in this case, after the outer `let` that assigns to `p`. This is
+illustrated in the example above by showing that `tmp` would be freed as we exit
+the block.
```

Comment:
There are four `let`s in this file, two in the first example and two in the second example. I think "after the `let`" refers to the first let in the first example, but you've updated this as if it's referring to the first let in the second example? Also, you've said "outer `let`" but there isn't an inner `let`.

I suggest just changing "after the `let`" to `after the `let p` in the first example above". What do you think?

Reply:
So if you count the `let tmp = foo();` on line 25 as one then there are five `let`s, and this one is the "inner `let`" that I was implying existed, but either way, I agree that adding in the "outer" wording is confusing since the second example is just used to explain what's going on in the first example. I think just making it read:
"after the `let p` statement"
would be sufficient because then you can refer back to either of the two examples and it's valid.

Reply:
:+1: